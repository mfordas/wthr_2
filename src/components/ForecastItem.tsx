import React, {useMemo} from 'react';
import {StyleSheet, View} from 'react-native';

import {Images} from 'app/assets';
import {types, constants} from 'app/constants';

import {ForecastProperty} from './ForecastProperty';

export interface Props {
  item: types.apiOneDayForecastResponse;
}

export const ForecastItem: React.FC<Props> = ({item}) => {
  console.log(item);
  const {temp, humidity, dt} = item;

  const date = useMemo(
    () => new Date(dt * constants.MILISECONDS_IN_SECOND),
    [dt],
  );

  return (
    <View style={styles.container}>
      <ForecastProperty
        value={date ? `${date.getDate()}/${date.getMonth() + 1}` : ''}
      />
      <ForecastProperty
        imgSrc={Images.Weather}
        value={temp.morn}
        unit={constants.CELSIUS_DEG}
      />
      <ForecastProperty
        imgSrc={Images.Sun}
        value={temp.day}
        unit={constants.CELSIUS_DEG}
      />
      <ForecastProperty
        imgSrc={Images.Night}
        value={temp.eve}
        unit={constants.CELSIUS_DEG}
      />
      <ForecastProperty
        imgSrc={Images.Humidity}
        value={humidity}
        unit={constants.HUMIDITY_UNIT}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
    flexDirection: 'row',
    justifyContent: 'center',
    alignContent: 'center',
  },
});
