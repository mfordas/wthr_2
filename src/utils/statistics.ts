import {max, min, mean, mode, round} from 'mathjs';

import {types} from 'app/constants';

export const prepareStatisticsData = (
  apiData: types.apiOneDayForecastResponse[],
) => {
  const maxTemperatures = apiData.map(day => round(day.temp.max));
  const minTemperatures = apiData.map(day => round(day.temp.min));

  return {maxTemperatures, minTemperatures};
};

export const countMaxValue = (arrayOfValues: number[]) =>
  round(max(arrayOfValues));

export const countMinValue = (arrayOfValues: number[]) =>
  round(min(arrayOfValues));

export const countModeValue = (arrayOfValues: number[]) => {
  const modeOfNumbers: number[] = mode(arrayOfValues);
  return modeOfNumbers.length === arrayOfValues.length ? false : modeOfNumbers;
};

export const countMeanValue = (arrayOfValues: number[]) =>
  round(mean(arrayOfValues));
