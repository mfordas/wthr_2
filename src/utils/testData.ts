import {types} from 'app/constants';

export const mockedGeolocationApiCityReponse: types.apiGeocodeCityResponse = {
  name: 'London',
  lat: 12.123456789,
  lon: 12.123456789,
  local_names: [
    {
      af: 'Londen',
    },
  ],
  country: 'GB',
};

export const mockedOneDayForecastApiResponse: types.apiOneDayForecastResponse =
  {
    clouds: 52,
    dew_point: 12.27,
    dt: 1655373600,
    feels_like: {
      day: 24.53,
      eve: 18.1,
      morn: 11.91,
      night: 13.84,
    },
    humidity: 46,
    moon_phase: 0.58,
    moonrise: 1655415960,
    moonset: 1655352300,
    pop: 0.98,
    pressure: 1017,
    rain: 3.43,
    sunrise: 1655346977,
    sunset: 1655406530,
    temp: {
      day: 24.79,
      eve: 18,
      max: 26.05,
      min: 12.47,
      morn: 12.47,
      night: 14.13,
    },
    uvi: 5.09,
    weather: [
      {
        description: 'moderate rain',
        icon: '10d',
        id: 501,
        main: 'Rain',
      },
    ],
    wind_deg: 310,
    wind_speed: 6.65,
  };
