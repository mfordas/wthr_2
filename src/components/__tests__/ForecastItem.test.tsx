import {render} from '@testing-library/react-native';
import React from 'react';

import {mockedOneDayForecastApiResponse} from 'app/utils/testData';

import {ForecastItem, Props} from '../ForecastItem';

const props: Props = {
  item: mockedOneDayForecastApiResponse,
};
const wrapper = render(<ForecastItem {...props} />);
test('ForecastItem should match snapshot', () => {
  expect(wrapper).toMatchSnapshot();
});
