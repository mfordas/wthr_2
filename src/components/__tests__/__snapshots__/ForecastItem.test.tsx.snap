// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastItem should match snapshot 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 68,
        "justifyContent": "center",
        "width": "20%",
      }
    }
  >
    <Text>
      16/6
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 68,
        "justifyContent": "center",
        "width": "20%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/weather.png",
        }
      }
      style={
        Object {
          "height": 24,
          "margin": 12,
          "width": 24,
        }
      }
    />
    <Text>
      12
      °C
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 68,
        "justifyContent": "center",
        "width": "20%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/sun.png",
        }
      }
      style={
        Object {
          "height": 24,
          "margin": 12,
          "width": 24,
        }
      }
    />
    <Text>
      25
      °C
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 68,
        "justifyContent": "center",
        "width": "20%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/night.png",
        }
      }
      style={
        Object {
          "height": 24,
          "margin": 12,
          "width": 24,
        }
      }
    />
    <Text>
      18
      °C
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 68,
        "justifyContent": "center",
        "width": "20%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/humidity.png",
        }
      }
      style={
        Object {
          "height": 24,
          "margin": 12,
          "width": 24,
        }
      }
    />
    <Text>
      46
      %
    </Text>
  </View>
</View>
`;
